Fixed #19878 -- Deprecated TemplateView passing URL kwargs into context.

adamchainz opened the pull request:
[Ticket #19878](https://code.djangoproject.com/ticket/19878)

adamchainz commented:
This is still a bit of WIP - I need to follow the deprecation guidelines and come up with a better message.

step21 commented:
Thanks. Let me know if you need help with something or documentation or whatever.

carltongibson commented on django/views/generic/base.py:
```diff
@@ -152,10 +155,24 @@ def get_template_names(self):
     """
     def get(self, request, *args, **kwargs):
-        context = self.get_context_data(**kwargs)
+        context_kwargs = {}
```
Can you put this block in a helper?

```
context_kwargs = _adams_helper_with_a_good_name(...)
```

adamchainz replied:
👌 Decided not to put my name in the function signature.

felixxm commented:
@adamchainz Is this ready for a review? (release notes are missing).

adamchainz commented:
Thanks for the bump @felixxm 

Came back to it and fixed a pretty vicious bug with the closure binding, hence moved the test to use two variables.

Added a release note.

felixxm commented:
@adamchainz Thanks :+1: I moved deprecation tests to a separate class, simplified them and pushed minor edits to the docs.

adamchainz commented:
All looks good 👍
